### PR TITLE
Improve command line parsing, adapt to Eyescale/CMake#533

### PR DESCRIPTION
**Pull request by rdumusc**

### CODE REVIEW

**Comment by rdumusc** [comment]
retest this please

File: apps/CMakeLists.txt [hunk 1]
```diff
@@ -1,25 +1,46 @@
...
 
 add_subdirectory(TideForker)
 add_subdirectory(TideMaster)
 add_subdirectory(TideWall)
 add_subdirectory(Launcher)
 add_subdirectory(Whiteboard)
+
+set(_tide_in ${CMAKE_CURRENT_SOURCE_DIR}/tide)
+set(_tide_out ${PROJECT_BINARY_DIR}/bin/tide)
+add_custom_command(
```
Comment (tribal-tec) [inline apps/CMakeLists.txt]: why not configure_file(@only)? This way it should run only if _tide_in has changed

Comment (rdumusc) [inline apps/CMakeLists.txt]: Good question, I had to investigate more to understand:
- configure_file(COPYONLY) is executed during CMake configure step, which means that CMake has to be re-run if the file has changed.
- also, if one does "rm bin/tide", then "ninja tide" doesn't know how to recreate the missing dependency (ninja: error: 'bin/tide', needed by 'apps/CMakeFiles/tide', missing and no known rule to make it).
- with the custom_command aproach the file is correctly recreated if it is missing, and it is also copied only if it has changed (I checked the exact modification dates).

However, my code was missing "DEPENDS ${_tide_in}" in add_custom_command(), so it did not copy the file again if _tide_in was changed. Now everything works as expected.

Comment (tribal-tec) [inline apps/CMakeLists.txt]: Ok, good then.

**Pull request merged**